### PR TITLE
fix some tests for Jetty 12 ee10 servlets

### PR DESCRIPTION
**Pull request by lachlan-roberts**
fix ee10 servlet tests associated with dispatching
fix some tests failing due to the Servlet 6 URI encoding changes

### CODE REVIEW

File: jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java [hunk 1]
```diff
@@ -669,7 +670,7 @@ public void setContextPath(String contextPath)
     {
         if (isStarted())
             throw new IllegalStateException(getState());
-        _contextPath = contextPath;
+        _contextPath = URIUtil.normalizePath(contextPath);
```
Comment (gregw) [inline jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java]: See #8343, as that will change this to canonicalizePath. ... race you to merge :)

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java [hunk 2]
```diff
@@ -134,11 +134,17 @@ public ServletApiResponse getServletApiResponse()
...
-        _httpOutput.reopen();
+        _httpServletResponse.resetBuffer();
         _outputType = OutputType.NONE;
     }
 
+    public void included()
+    {
+        if (_outputType == OutputType.WRITER)
+            _writer.reopen();
+        _httpOutput.reopen();
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java:145]: Why is this needed?  I thought the include response wrapper intercepts close, so it cannot be closed?
Is this for any sendErrors within an incliude?  A comment explaining would be good.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletHandler.java [hunk 3]
```diff
@@ -775,7 +775,7 @@ public void addServletWithMapping(ServletHolder servlet, String pathSpec)
 
             ServletMapping mapping = new ServletMapping();
             mapping.setServletName(servlet.getName());
-            mapping.setPathSpec(pathSpec);
+            mapping.setPathSpec(URIUtil.normalizePath(pathSpec));
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletHandler.java]: will be renamed to canonicalPath after #8343

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletHandler.java [hunk 4]
```diff
@@ -887,7 +887,7 @@ public void addFilterWithMapping(FilterHolder holder, String pathSpec, EnumSet<D
 
             FilterMapping mapping = new FilterMapping();
             mapping.setFilterName(holder.getName());
-            mapping.setPathSpec(pathSpec);
+            mapping.setPathSpec(URIUtil.normalizePath(pathSpec));
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletHandler.java]: ditto

**Review by gregw — CHANGES_REQUESTED** [review]
I think we need to ponder our space handling for a little bit.
The servlet 6.0 spec does allow us to return some encoded characters in servletPath and pathInfo, but that is primarily intended for things like `%2F`.

We are currently leaving `%20` encoded because the JVM URI class barfs on space as an illegal character.  But perhaps we should decode for the servlet  API and then ensure we re-encode in the API - which is something that #8343 has added to some extent.

So whilst we ponder, can you add additional tests to this PR that test encoded %2F and %3F in servlet patterns.  That way we can later change our %20 handling and still test the encoding this PR is doing.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java (hunk 2 above)
Comment (lachlan-roberts) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java:145]: I was just replicating all the places where this was called from Jetty 11. So maybe this is no longer needed. But then this will just be a NOOP anyway.

**Review by joakime — APPROVED** [review]
I like it!

I'm anxious to get my hands on these changes for the various DefaultServlet tests i'm currently working on too.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java (hunk 2 above)
Comment (gregw) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java:145]: Be wary of just copying behavours from jetty-11 as much as changed around things like dispatch and content management.  So good to question anything that looks suspect.
I'm guessing it is for sendErrors... but good to verify that it does do something sometimes and to then add a comment.

**Review by gregw — APPROVED** [review]
Since this is blocking other work, I will approve.
But can you investigate if the reopen after included really is needed and add a comment explaining it (no need for PR).
